vm/devices/vmbus/vmbus_server/src/channels.rs

```diff
@@ -2590,53 +2592,63 @@ impl<'a, N: 'a + Notifier> ServerWithNotifier<'a, N> {
         input: &protocol::GpadlHeader,
         range: &[u8],
     ) -> Result<(), ChannelError> {
-        // Validate the channel ID.
-        let (offer_id, channel) = self
-            .inner
-            .channels
-            .get_by_channel_id_mut(&self.inner.assigned_channels, input.channel_id)?;
+        (|| {
```

Comment:
I'd suggest splitting this into a separate named function rather than do this closure trick.